cmd/otel-allocator/main.go

```diff
@@ -220,17 +224,32 @@ func (s *server) Shutdown(ctx context.Context) error {
 // After that, the YAML is converted in to a JSON format for consumers to use.
 func (s *server) ScrapeConfigsHandler(w http.ResponseWriter, r *http.Request) {
 	configs := s.discoveryManager.GetScrapeConfigs()
-	configBytes, err := yaml2.Marshal(configs)
+
+	hash, err := hashstructure.Hash(configs, nil)
 	if err != nil {
+		s.logger.Error(err, "failed to hash the config")
 		s.errorHandler(w, err)
+		return
 	}
-	jsonConfig, err := yaml.YAMLToJSON(configBytes)
-	if err != nil {
-		s.errorHandler(w, err)
+	// if the hashes are different, we need to recompute the scrape config
```

Comment:
this logic is similar to what the collector is doing on the receiving end i.e. it should only update this byte array when there's an updated scrape config list